data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/PrometheusSink.java

```diff
@@ -4,116 +4,96 @@
 
-    private static final Logger LOG = LoggerFactory.getLogger(PrometheusSink.class);
-
-    private static final String BUCKET = "bucket";
-    private static final String KEY_PATH = "key_path_prefix";
-
     private volatile boolean sinkInitialized;
-
     private final PrometheusSinkService prometheusSinkService;
-
-    private DlqPushHandler dlqPushHandler;
+    private PrometheusHttpSender httpSender;
+    private SinkMetrics sinkMetrics;
 
     @DataPrepperPluginConstructor
     public PrometheusSink(final PluginSetting pluginSetting,
-                    final PrometheusSinkConfiguration prometheusSinkConfiguration,
+                    final PluginMetrics pluginMetrics,
                     final PluginFactory pluginFactory,
-                    final PipelineDescription pipelineDescription,
+                    final PrometheusSinkConfiguration prometheusSinkConfiguration,
                     final AwsCredentialsSupplier awsCredentialsSupplier) {
         super(pluginSetting);
         this.sinkInitialized = Boolean.FALSE;
-        if (prometheusSinkConfiguration.getDlqFile() != null) {
-            this.dlqPushHandler = new DlqPushHandler(prometheusSinkConfiguration.getDlqFile(), pluginFactory,
-                    null, null, null, null, pluginMetrics);
-        }else {
-            this.dlqPushHandler = new DlqPushHandler(prometheusSinkConfiguration.getDlqFile(), pluginFactory,
-                    String.valueOf(prometheusSinkConfiguration.getDlqPluginSetting().get(BUCKET)),
-                    prometheusSinkConfiguration.getDlqStsRoleARN()
-                    , prometheusSinkConfiguration.getDlqStsRegion(),
-                    String.valueOf(prometheusSinkConfiguration.getDlqPluginSetting().get(KEY_PATH)), pluginMetrics);
-        }
-        final HttpRequestRetryStrategy httpRequestRetryStrategy = new DefaultHttpRequestRetryStrategy(prometheusSinkConfiguration.getMaxUploadRetries(),
-                TimeValue.of(prometheusSinkConfiguration.getHttpRetryInterval()));
-
-        if((!prometheusSinkConfiguration.isInsecure()) && (prometheusSinkConfiguration.isHttpUrl())){
-            throw new InvalidPluginConfigurationException ("Cannot configure http url with insecure as false");
-        }
+        AwsConfig awsConfig = prometheusSinkConfiguration.getAwsConfig();
+        final AwsCredentialsProvider awsCredentialsProvider = (awsConfig != null) ? awsCredentialsSupplier.getProvider(convertToCredentialOptions(awsConfig)) : awsCredentialsSupplier.getProvider(AwsCredentialsOptions.builder().build());
+        Region region = (awsConfig != null) ? awsConfig.getAwsRegion() : awsCredentialsSupplier.getDefaultRegion().get();
+      
+        sinkMetrics = new DefaultSinkMetrics(pluginMetrics, "metric");
+        httpSender = new PrometheusHttpSender(awsCredentialsSupplier, prometheusSinkConfiguration, sinkMetrics, prometheusSinkConfiguration.getConnectionTimeoutMs(), prometheusSinkConfiguration.getIdleTimeoutMs());
```

Comment:
Let's start moving toward dependency injection. This is support now. You can see an example of this in PR #6190. I added DI to the DynamoDB source coordinator.

Reply:
I will do this in a fast-followup PR